### PR TITLE
Refactored app architecture

### LINKED ISSUE #20: Refactor App to not perform data fetching
Architecture could be improved by not performing data fetching within `App`. It shouldn't be responsible for fetching discussion data, self, other data for other views...

Instead `App` should simply instantiate child views that are responsible for fetching the data they require. This will also make it less messy for students to build new views - they'll simply delete components they don't need instead of having to also delete the `useEffects` within `App`. 

Might also be worth creating a `Welcome` component to encapsulate the welcome message that performs the `getSelf` fetch. 

Documentation also needs to be updated to reflect these changes.

### PR DESCRIPTION
**Pull request by justin0022**
Resolves #20 

* refactored `App` to not fetch data
* created `Welcome` component
* moved data fetching `useEffect` hints from `App` to `Welcome` and `Discussion`.

Need to update documentation to reflect these changes.

### CODE REVIEW

**Comment by justin0022** [comment]
Just realized I need to make some further changes, please don't review until I ping you

**Comment by justin0022** [comment]
Should be ready for review now.

File: 4-Create-Data-Viz-Dashboard.md [hunk 1]
```diff
@@ -5,44 +5,41 @@
...
+useEffect(() => {
+  if (discussion.length > 0) {
+    const discussionTimestamps = discussion
+      .map(discussion => discussion.timestamp)
+
+    setTimestamps(discussionTimestamps)
+  }
+}, [discussion])
 ```
+It checks to see if discussion data has been set - if so, it extracts the timestamp property and sets the `timestamps` variable. Why do this instead of just sending the data as is to `Heatmap`? Well, we designed `Heatmap` to only rely on `timestamps`, and it's better practice to send child components just the data it requires instead of the entire kitchen sink.
```
Comment (markoprodanovic) [inline 4-Create-Data-Viz-Dashboard.md:25]: I like that you've explained and encouraged best practices! Also forces participants to stop and really think about what's going on 👍

**Review by markoprodanovic — APPROVED** [review]
Looks good to me!

**Comment by justin0022** [comment]
@allcontributors please add @markoprodanovic as a contributor for doc, ideas, code, eventOrganizing, review.

**Comment by justin0022** [comment]
@allcontributors please add @markoprodanovic as a contributor for doc, ideas, code, eventOrganizing, review.

**Pull request merged**